daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs

```diff
@@ -228,7 +232,12 @@ runStart
     jsonApiOpts <- withOptsFromProjectConfig "json-api-options" jsonApiOpts projectConfig
     scriptOpts <- withOptsFromProjectConfig "script-options" scriptOpts projectConfig
     doBuild
-    when generateJsCode $ doJsCodegen [darPath, "-o", "ui/daml.js"]
+    forM_ [minBound :: Lang .. maxBound :: Lang] $ \lang -> do
+      mbOutputPath :: Maybe String <-
+        requiredE ("Failed to parse codegen entry for " <> showLang lang) $
+        queryProjectConfig ["codegen", showLang lang] projectConfig
```

Comment:
I think the `js` codegen is the only one that you need to pass arguments from `daml.yaml` explicitly for now. AFAIK the java and scala codegens pick up options from daml.yaml automatically (I don't know when).

See for example the codegen option that's in the `daml.yaml` for the quickstart-java template:

```
codegen:
  java:
    package-prefix: com.daml.quickstart.model
    output-directory: target/generated-sources/iou
    decoderClass: com.daml.quickstart.iou.TemplateDecoder
```

And for the quickstart-scala template:

```
codegen:
  scala:
    package-prefix: com.daml.quickstart.iou.model
    output-directory: scala-codegen/src/main/scala
    verbosity: 2
```

(Probably the right thing to do, as Moritz said, is to fix the js codegen so it can pick up arguments from daml.yaml automatically... But I think that can be done separately.)

Reply:
yes that's true, I just changed it when I discovered it in the codegen docs.